### PR TITLE
[TTAHUB-502] Add AWS Elasticsearch text search filter

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

**NOTE:** This contains required changes from https://github.com/HHS/Head-Start-TTADP/pull/1250 also under review.

This add a 'Text' filter to the AR landing to allow the search of the 'activityreport' index documents.

image

Keep in mind that this searches reports that have been indexed via a submit or the indexing job (approved reports).

It currently searches the following fields:

- context
- recipient next steps
- specialist next steps
- goal text
- objective text
- objective tta provided

It may also return results that contain **partial matches**.

## How to test

_**Prerequisites:**_ Before testing you should run the index script '**createAwsElasticSearchIndexes**'. Once completed you will need to make sure both **DOCKER** and the **WORKER** are running.

To start I would recommend creating and submitting a report that has unique values for each of the above fields. Then create 'Text' filter on the AR landing and make sure you get the desired results. Also test this same filter in conjunction with other filters.

Finally test the new filter against the existing reports. 

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-502

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

File: src/scopes/index.js [hunk 1]
```diff
@@ -1,13 +1,38 @@
...
+import { AWS_ELASTIC_SEARCH_INDEXES, DECIMAL_BASE } from '../constants';
+import { search } from '../lib/awsElasticSearch/index';
 
 const models = {
   activityReport,
   grant,
   goal,
 };
 
+async function checkForSearchItems(filters) {
```
Comment (AdamAdHocTeam) [inline src/scopes/index.js:15]: Putting this here for now might make sense in the future as this grows to rip it out into its own file.

**Comment by thewatermethod** [comment]
Would prefer to review this after #1250 is merged into ```main``` as it'll shrink the size of this PR a bit

**Comment by thewatermethod** [comment]
This works really well! Awesome

Is this something that we can add a e2e test for?

**Comment by AdamAdHocTeam** [comment]
> This works really well! Awesome
> 
> Is this something that we can add a e2e test for?

Yes it will be written next for https://ocio-jira.acf.hhs.gov/browse/TTAHUB-873.

**Review by thewatermethod — APPROVED** [review]

**Pull request merged**